### PR TITLE
Intelligent time limits

### PR DESCRIPTION
**Pull request by 7maples**
[Pivotal Card](https://www.pivotaltracker.com/story/show/91605682)

Assigned @tank157

### CODE REVIEW

File: src/usps_processor/api.clj [hunk 1]
```diff
@@ -51,26 +52,52 @@
...
+      (assoc :status 422)))
+
+(defn get-scanned-since [req]
+  (when-let [string (get-in req [:params :scanned-since])]
+    (instant/read-instant-date string)))
 
 (defn latest-scan [req]
-  (on-single-match (lookup-mailings req)
-                   (comp render-scan mailing/latest-scan)))
+  (let [matches (lookup-mailings req)]
```
Comment (tank157) [inline src/usps_processor/api.clj]: I think you want to apply the time filtering to matches here before going into the conditional. It should be ok for there to be multiple matches in general if all but one are before the time cut-off, for example.

File: src/usps_processor/api.clj [hunk 2]
```diff
@@ -51,26 +52,52 @@
...
+
+       :else
+       (if-let [the-scan (mailing/latest-scan-since (first matches) (get-scanned-since req))]
+         (edn-response (render-scan the-scan))
+         standard-not-found))))
 
 (defn all-scans [req]
-  (on-single-match (lookup-mailings req)
-                   (comp (partial map render-scan) mailing/all-scans)))
+  (let [matches (lookup-mailings req)]
```
Comment (tank157) [inline src/usps_processor/api.clj]: Same comment about filtering time first applies here as well.

File: test/usps_processor/api_test.clj [hunk 3]
```diff
@@ -1,7 +1,30 @@
...
+                                                      {:scan/time #inst "1999-01-01" :db/id scan1
+                                                       :scan/mailing mailing-id}
+                                                      {:scan/time #inst "2000-01-01" :db/id scan2
+                                                       :scan/mailing mailing-id}
+                                                      {:scan/time #inst "2001-01-01" :db/id scan3
+                                                       :scan/mailing mailing-id}]))
+                      (test)
+                      (db/delete-database (config :datomic :uri))))
+
+(comment
```
Comment (tank157) [inline test/usps_processor/api_test.clj]: commented this out? If it's invalid perhaps we should remove it? Or if not, add it back in.

**Comment by tank157** [comment]
Otherwise LGTM, feel free to merge when you resolve the println and commented out test

**Comment by ericnormand** [comment]
Tests pass locally. Ready to merge!

**Pull request merged**